lib/chef/resource/windows_firewall_rule.rb

```diff
@@ -139,10 +172,17 @@ class WindowsFirewallRule < Chef::Resource
       action :create do
         description "Create a Windows firewall entry."
 
+        unless is_set_properly?(new_resource.icmp_type, new_resource.protocol)
```

Comment:
We have a method `define_resource_requirements` that gets called at the very beginning, which I think handles the scenario you're hitting here. Here's how I used it to validate inputs per action in the `alternatives` resource.

https://github.com/chef/chef/blob/master/lib/chef/resource/alternatives.rb#L37

Reply:
Will definitely look into this. Thanks a lot!